lib/Sema/TypeCheckStorage.cpp

```diff
@@ -354,6 +354,36 @@ InitAccessorPropertiesRequest::evaluate(Evaluator &evaluator,
   return decl->getASTContext().AllocateCopy(results);
 }
 
+ArrayRef<VarDecl *>
+MemberwiseInitPropertiesRequest::evaluate(Evaluator &evaluator,
+                                        NominalTypeDecl *decl) const {
+  IterableDeclContext *implDecl = decl->getImplementationContext();
+
+  if (!hasStoredProperties(decl, implDecl))
+    return ArrayRef<VarDecl *>();
+
+  // Make sure we expand what we need to to get all of the properties.
+  computeLoweredProperties(decl, implDecl, LoweredPropertiesReason::Memberwise);
+
+  SmallVector<VarDecl *, 4> results;
+  auto maybeAddProperty = [&](VarDecl *var) {
+    if (var->isMemberwiseInitialized(/*preferDeclaredProperties=*/true))
+      results.push_back(var);
+  };
+
+  for (auto *member : decl->getCurrentMembers()) {
+    if (auto *var = dyn_cast<VarDecl>(member))
+      maybeAddProperty(var);
+
+    member->visitAuxiliaryDecls([&](Decl *auxDecl) {
+      if (auto auxVar = dyn_cast<VarDecl>(auxDecl))
+        maybeAddProperty(auxVar);
+    });
```

Comment:
If I understand this correctly there might be a problem with the way this is set up because `init accessors` subsume initialization of some stored properties which means that loop can discover a stored property and add it to results but later discover a property with an `init accessor` (i.e. added by macro expansion) that should actually be used instead. Currently `createImplicitConstructor` establishes mappings between stored properties and their init accessor initializing properties upfront before iterating over the members (assuming that expansions already happened which might not be correct).

Reply:
I think we should integrate that to this method and let `createImplicitConstructor` just iterate over the properties instead.

Reply:
`computeLoweredProperties` forces everything to be expanded that will be expanded. The just-pushed https://github.com/apple/swift/pull/66867/commits/9ea6dcac76134fa488f885273a0288f73433814b does the filtering out of stored properties that are subsumed by computed properties with init accessors, as you're suggesting, so  `createImplicitConstructor` doesn't need special logic.

Reply:
Looks good thank you!